addons/website_sale_options/controllers/main.py

```diff
@@ -55,21 +43,18 @@ def cart_options_update_json(self, product_id, add_qty=1, set_qty=0, goto_shop=N
 
     @http.route(['/shop/modal'], type='json', auth="public", methods=['POST'], website=True)
     def modal(self, product_id, **kw):
-        cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
         pricelist = request.website.get_current_pricelist()
-        if not context.get('pricelist'):
-            context['pricelist'] = int(pricelist)
+        ctx = request.context
+        if not ctx.get('pricelist'):
+            ctx['pricelist'] = int(pricelist)
 
         website_context = kw.get('kwargs', {}).get('context', {})
-        context = dict(context or {}, **website_context)
-        from_currency = pool['res.users'].browse(cr, uid, uid, context=context).company_id.currency_id
+        from_currency = request.env.user.company_id.currency_id
         to_currency = pricelist.currency_id
-        compute_currency = lambda price: pool['res.currency']._compute(cr, uid, from_currency, to_currency, price, context=context)
-        product = pool['product.product'].browse(cr, uid, int(product_id), context=context)
-        request.website = request.website.with_context(context)
-
-        return request.website._render("website_sale_options.modal", {
-                'product': product,
-                'compute_currency': compute_currency,
-                'get_attribute_value_ids': self.get_attribute_value_ids,
-            })
+        compute_currency = lambda price: request.env['res.currency']._compute(from_currency, to_currency, price)
+        product = request.env['product.product'].browse(int(product_id))
+        return request.website.with_context(dict(ctx, **website_context))._render("website_sale_options.modal", {
```

Comment:
@rga-odoo why do you use the custom context for the `_render` call? Where is it used?